Zephyr Quick Track Cleanup

D-Triveni opened the pull request.

D-Triveni committed:
[nrf noup] zephyr: Add dependent kconfig

fixup! [nrf noup] zephyr: Add support for Zephyr

To include posix reated API's.
Need to be enabled by application.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup] zephyr: qt: Add missing main and other dependent file

Add boot time thread to start QT.
Added option to select the log level.

fixup! [nrf noup] zephyr: Add support for Zephyr

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup] zephyr: Remove OpenWRT code

Cleanup OpenWRT code from zephyr specific files.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup] zephyr: Remove File Operations code

Clean up the file operation based functionlaity,
as there is no support.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

krish2718 commented on eloop.c:
```diff
@@ -20,7 +20,11 @@
 #include <errno.h>
 #include <signal.h>
+#ifdef CONFIG_ZEPHYR
+#include <zephyr/posix/sys/select.h>
```
Please put this in a separate commit explaining why we need this?

krish2718 commented on eloop.c:
```diff
@@ -283,9 +287,9 @@ static void eloop_process_pending_signals(void)
 
 	if (eloop.pending_terminate) {
-#ifndef CONFIG_NATIVE_WINDOWS
+#if !defined(CONFIG_NATIVE_WINDOWS) && !defined(CONFIG_ZEPHYR)
```
Please put this in a separate commit explaining why we need this?

krish2718 commented on zephyr/src/dut_main.c:
```diff
@@ -0,0 +1,249 @@
+/* Copyright (c) 2020 Wi-Fi Alliance                                                */
```
Instead of copying OSS code, can we leverage those functions here and keep this file Zephyr specific (with nRF license)?

krish2718 reviewed (changes requested):
I am still thinking about keeping non-zephyr code under a define (which never gets defined) vs. removing it? Esp. from maintenance PoV, if we fork it and customize heavily then keeping it up-to-date across upmerge's will get tricky.

Lets address other comments and circle back to this.

D-Triveni committed:
[nrf noup]: qt: Fix header file path

Required for select() call.

fixup! [nrf noup] zephyr: Add support for Zephyr

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: qt: Remove alarm call

fixup! [nrf noup] zephyr: Add support for Zephyr

Alarm's are not supported in Zephyr, see [1].

[1] - https://docs.zephyrproject.org/latest/services/portability/posix.html

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf fromlist]: Move socket based operation into new file

Add new file for socket opeartions, for common use
across different OS.

Upstream-Pr: https://github.com/Wi-FiQuickTrack/Wi-FiQuickTrack-ControlAppC/pull/6

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

krish2718 commented on eloop.c:
```diff
@@ -287,9 +287,9 @@ static void eloop_process_pending_signals(void)
 
 	if (eloop.pending_terminate) {
-#ifndef CONFIG_NATIVE_WINDOWS
+#if !defined(CONFIG_NATIVE_WINDOWS) && !defined(CONFIG_ZEPHYR)
```
If this commit is fixing another `nrf noup` please follow the procedure for that.

D-Triveni replied:
@krish2718 
After adding dut initialization changes(dut_main.c), and I am seeing compilation issues. So, I have added this change. 
But I committed this change before and later  dut initialization changes. So i feel no fixup is required.
And also, let me know is it okay if commit order is like this.

krish2718 replied:
whichever commit we start seeing compilation failure, this fix should be part of it.

D-Triveni replied:
InitIally I did that, u commented to split.

krish2718 replied:
Sorry, that looked like an unrelated fix, the commit log didn't explain, hence the confusion.

krish2718 replied:
so, it is fixing a `nrf noup`, so, you can keep this commit but follow the squashing policy.

krish2718 replied:
Your latest changes have overwritten mine, this and previous commit are missing the `fixup! [nrf noup]` line.

krish2718 commented on Makefile:
```diff
@@ -5,7 +5,7 @@ ROLE = dut
 VERSION = "2.2.0.46"
 
-OBJS = main.o eloop.o indigo_api.o indigo_packet.o utils.o wpa_ctrl.o
+OBJS = main.o eloop.o indigo_api.o indigo_packet.o utils.o wpa_ctrl.o qt_client.o
```
missing sauce tag for this commit.

krish2718 commented on zephyr/src/main.c:
```diff
@@ -0,0 +1,107 @@
+    register_apis();
+
+    /* Initiate the vendor's specific startup commands */
+    vendor_init();
```
Remove

krish2718 commented on zephyr/src/main.c:
```diff
@@ -0,0 +1,107 @@
+#include <zephyr/posix/pthread.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(wfa_qt, LOG_LEVEL_DBG);
```
Define a separate logger and use

D-Triveni replied:
Cant'we use this? Any specific reason for new logger?

krish2718 replied:
Yes, this is too generic and could enable other modules logs too. We need an option to enable logs only for this module.

D-Triveni replied:
@krish2718 Added, please review.

krish2718 commented on zephyr/src/main.c:
```diff
@@ -0,0 +1,107 @@
+
+#include <stdio.h>
+
+#include "vendor_specific.h"
```
no need

krish2718 commented on zephyr/src/main.c:
```diff
@@ -0,0 +1,107 @@
+                0);
+
+static pthread_t main_thread;
+#define STACK_SIZE 4096
```
remove

krish2718 commented on zephyr/src/main.c:
```diff
@@ -0,0 +1,107 @@
+
+static pthread_t main_thread;
+#define STACK_SIZE 4096
+K_THREAD_STACK_DEFINE(main_thread_stack, STACK_SIZE);
```
Use WFA_QT_THREAD_STACK_SIZE, see my comment about two threads below.

krish2718 commented on zephyr/src/main.c:
```diff
@@ -0,0 +1,107 @@
+        return;
+    }
+
+    ret = pthread_create(&main_thread, &ptAttr, main_thread_handler, NULL);
```
Why do you need a second thread here? Stick to Zephyr native APIs for Zephyr code.

D-Triveni replied:
Initially when QT isn't running from boot time, did this change to run app on a thread.
But now, we are running QT from boot time as a thread. So, no need for 2nd thread, will remove it and test.

krish2718 commented on zephyr/src/vendor_specific_dut.c:
```diff
@@ -23,329 +23,68 @@
 #include "vendor_specific.h"
 #include "utils.h"
 
-#ifdef HOSTAPD_SUPPORT_MBSSID_WAR
```
Just remove this file. In a generic code it makes sense to call vendor specific, but this is Zephyr specific already, doesn't make sense, unless we use generic `indigo_*` files and implement `vendor_*` files for Zephyr.

D-Triveni replied:
Based on discussion, keeping this file.

krish2718 commented on zephyr/src/vendor_specific_dut.c:
```diff
@@ -354,150 +93,54 @@ int get_p2p_dev_if(char *if_name, size_t size) {
-            return NULL;
-        }
-    }
+    /*TODO:
```
Can you use a single line for all `TODO`?

D-Triveni committed:
[nrf noup] zephyr: qt: Update License

Update the license year.

squash! [nrf noup] zephyr: Add support for Zephyr.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

krish2718 commented on zephyr/CMakeLists.txt:
```diff
@@ -26,10 +26,12 @@ zephyr_include_directories_ifdef(CONFIG_WFA_QT_CONTROL_APP
 	${SOURCES_BASE}/eloop.c
 	${SOURCES_BASE}/indigo_api.c
 	${SOURCES_BASE}/indigo_packet.c
+	${SOURCES_BASE}/qt_client.c
```
Commit title needs to cover all changes

krish2718 commented on zephyr/src/indigo_api_callback_dut.c:
```diff
@@ -1,4 +1,4 @@
-/* Copyright (c) 2020 Wi-Fi Alliance                                                */
+/* Copyright (c) 2023 Wi-Fi Alliance                                                */
```
we cannot change original year here as we don't own this copyright. We could add below, 

`/* Copyright (c) 2023 Nordic semiconductor ASA */`

krish2718 commented on zephyr/src/indigo_api_callback_dut.c:
```diff
@@ -14,6 +14,8 @@
 /* OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS                          */
 /* SOFTWARE. */
 
+/* Copyright (c) 2023 Nordic semiconductor ASA */
```
you can add this next to the first copyright.

krish2718 commented on zephyr/Kconfig:
```diff
@@ -7,4 +7,5 @@
 config WFA_QT_CONTROL_APP
 	# Need full POSIX from libc, Zephyr's POSIX support is only partial
 	depends on !POSIX_API
+	select PTHREAD_IPC
```
This commit log also needs `fixup!`

krish2718 commented on zephyr/Kconfig:
```diff
@@ -8,4 +8,11 @@ config WFA_QT_CONTROL_APP
+        int "Thread stack size"
+        default 4096
+        help
+          Thread stack size.
```
Remove this trivial and duplicate help or add a detailed help.

krish2718 commented on zephyr/Kconfig:
```diff
@@ -8,4 +8,11 @@ config WFA_QT_CONTROL_APP
 	bool "WFA Quicktrack control app"
+
+config WFA_QT_THREAD_STACK_SIZE
+        int "Thread stack size"
```
WFA QT Thread stack size
